app/models/moderation_report.rb

```diff
@@ -0,0 +1,57 @@
+class ModerationReport < ApplicationRecord
+  belongs_to :model, polymorphic: true
+  belongs_to_creator
```

Comment:
The model should have a `validates :reason, presence: true` validation to ensure that the reason can't be blank.